src/features/profile/components/Categories.tsx

```diff
@@ -0,0 +1,76 @@
+import React, { useState, useRef } from 'react'
+import ico_add from '../../../common/assets/icons/ico-add.svg'
+import ico_close from '../../../common/assets/icons/ico-close-round.svg'
+
+export type TCategories = {
+  value: Array<string>
+  onChange: React.Dispatch<React.SetStateAction<Array<string>>>
+}
+
+const Categories = ({ value, onChange }: TCategories): JSX.Element => {
+  const [isAdding, setAdding] = useState<boolean>(false)
+  const inputRef = useRef<HTMLDivElement>(null)
+
+  const onAddCategory = (e: React.KeyboardEvent<HTMLDivElement>) => {
+    if (e.key == 'Enter') {
+      const newCategory = inputRef?.current?.innerHTML
+      if (!newCategory) {
+        return
+      }
+      const newValue = [...value]
+      newValue.push(newCategory)
+      onChange(newValue)
+      setAdding(false)
+    }
+  }
+
+  const onDeleteCategory = (index: number) => {
+    const newValue = [...value]
+    newValue.splice(index, 1)
+    onChange(newValue)
+  }
+  return (
+    <div className='bg-white border rounded flex flex-grow shadow-editbox flex-wrap p-2 items-center'>
+      {value.map((category: string, index: number) => (
+        <div className='flex' key={index}>
+          <div className='px-2 mx-2 rounded-full text-gray-dd border border-gray-dd flex text-sm items-center  mb-2'>
+            <div className='h-6 text-gray-4a'>{category}</div>
+            <div
+              className='bg-gray-e6 ml-1 rounded-full h-4 w-4 flex justify-center items-center'
+              onClick={() => onDeleteCategory(index)}
+            >
+              <img src={ico_close} className='cursor-pointer' />
+            </div>
+          </div>
+        </div>
+      ))}
+      {!isAdding && (
+        <img
+          src={ico_add}
+          onClick={() => setAdding(true)}
+          className='cursor-pointer w-21px mb-2'
+        />
+      )}
+      {isAdding && (
+        <div className='flex'>
+          <div className='px-2 mx-2 rounded-full text-gray-dd border border-gray-dd flex text-sm items-center  mb-2'>
+            <div
+              onKeyPress={onAddCategory}
+              ref={inputRef}
+              className='h-6 text-gray-4a min-w-32px outline-none'
+              contentEditable={true}
+            />
```

Comment:
@CrazyOptimist , If I use input here, it can't be resizable according to its content and it will look ugly.

Reply:
contentEditable is only for the case when we want to store html user input, for rich text editor for example. As with contentEditable user can submit raw html, it requires special treatment - whitelisting of allowed tags, removing styles, etc.

We have existing dependency "downshift" for creating customizable select inputs:
https://codesandbox.io/s/o4yp9vmm8z

Style and behavior of this library is fully customizable. It may be complicated, so let me know if you would need any help